Homework2/Trie/Trie.cs

```diff
@@ -0,0 +1,260 @@
+            ++currentElement!.SizeOfTrieElement;
+            ++stringElementIndex;
+        }
+
+        return currentElement!.IsTerminal = true;
+    }
+
+    public void AddWithPointer(char newElement)
+    {
+        TrieElement? currentElement = this.triePointer;
+        if (currentElement!.Vertexes.ContainsKey(newElement))
+        {
+            currentElement = currentElement.Vertexes[newElement] as TrieElement;
+            if (!currentElement!.IsTerminal)
+            {
+                currentElement!.IsTerminal = true;
+                ++currentElement.SizeOfTrieElement;
+                ++this.head.SizeOfTrieElement;
+                currentElement.VertexValue = this.head.SizeOfTrieElement + 255;
+                this.triePointer = this.head;
+            }
+
+            this.triePointer = (TrieElement?)currentElement.Vertexes[newElement];
+        }
+
+        currentElement.Vertexes[newElement] = new TrieElement();
+        currentElement = currentElement.Vertexes[newElement] as TrieElement;
+        currentElement!.IsTerminal = true;
+        ++currentElement.SizeOfTrieElement;
+        ++this.head.SizeOfTrieElement;
+        currentElement.VertexValue = this.head.SizeOfTrieElement + 255;
+        this.triePointer = this.head;
+    }
+
+    /// <summary>
+    /// Method checks the element's existence in Trie.
+    /// </summary>
+    /// <param name="stringElement">The string type element to check.</param>
+    /// <returns>True if element contains in Trie. Else false.</returns>
+    public bool Contains(string stringElement)
+    {
+        if (stringElement.Length == 0 || stringElement == null)
+        {
+            return false;
+        }
+
+        var stringElementIndex = 0;
+        TrieElement? currentElement = this.head;
+        while (currentElement != null && currentElement.Vertexes.ContainsKey(stringElement[stringElementIndex]))
+        {
+            currentElement = currentElement.Vertexes[stringElement[stringElementIndex]] as TrieElement;
+            if (stringElementIndex == stringElement.Length - 1)
+            {
+                return currentElement!.IsTerminal;
+            }
+
+            ++stringElementIndex;
+        }
+
+        return false;
+    }
+
+    /// <summary>
+    /// Method removes the element from the Trie.
+    /// </summary>
+    /// <param name="stringElement">The string type element to removing.</param>
+    /// <returns>True if element has been contained in the Trie. Else false.</returns>
+    public bool Remove(string stringElement)
+    {
+        if (!this.Contains(stringElement) || stringElement.Length == 0 || stringElement == null)
+        {
+            return false;
+        }
+
+        var stringElementIndex = 0;
+        TrieElement? currentElement = this.head;
+
+        var elementWithLastTermainalPointer = currentElement.Vertexes[stringElement[0]] as TrieElement;
```

Comment:
```suggestion
        var elementWithLastTerminalPointer = currentElement.Vertexes[stringElement[0]] as TrieElement;
```
Только он потом всё равно не пригодился